src/implementations/twig/components/accordion/accordion.story.js

```diff
@@ -1,40 +1,53 @@
-import { withKnobs, text, select, optionsKnob } from '@storybook/addon-knobs';
 import { withNotes } from '@ecl/storybook-addon-notes';
-import { getExtraKnobs, tabLabels } from '@ecl/story-utils';
 import withCode from '@ecl/storybook-addon-code';
 import defaultSprite from '@ecl/resources-ec-icons/dist/sprites/icons.svg';
 import demoData from '@ecl/specs-component-accordion/demo/data';
 
 import accordion from './accordion.html.twig';
 import notes from './README.md';
 
-const prepareAccordion = (data) => {
-  data.items.forEach((item, index) => {
-    const levels = [1, 2, 3, 4, 5, 6];
-    const { id, level, toggle, content } = item;
-    item.id = text(`items[${index}].id`, id, tabLabels.required);
-    item.content = text(`items[${index}].content`, content, tabLabels.required);
-    item.toggle.label = text(
-      `items[${index}].toggle.label`,
-      toggle.label,
-      tabLabels.required
-    );
-    item.toggle.icon.path = optionsKnob(
-      `items[${index}].toggle.icon.path`,
-      { current: defaultSprite, 'no path': '' },
-      defaultSprite,
-      { display: 'inline-radio' },
-      tabLabels.required
-    );
-    item.level = select(
-      `items[${index}].level`,
-      levels,
-      level,
-      tabLabels.optional
-    );
+const getArgTypes = () => {
+  const argTypes = {};
+  demoData.items.forEach((item, i) => {
+    argTypes[`toggle${i + 1}`] = {
+      name: `toggle ${i + 1}`,
+      type: { name: 'string', required: true },
+      defaultValue: item.toggle.label,
+      description: `Text of the accordion toggler`,
+      table: {
+        type: { summary: 'string' },
+        defaultValue: { summary: '' },
+        category: 'Toggles',
+      },
+      control: {
+        type: 'text',
+      },
+    };
+    argTypes[`content${i + 1}`] = {
+      name: `content ${i + 1}`,
+      type: { name: 'string', required: true },
+      defaultValue: item.content,
+      description: 'Text of the hidden content',
+      table: {
+        type: { summary: 'string' },
+        defaultValue: { summary: '' },
+        category: 'Hidden content',
+      },
+      control: {
+        type: 'text',
+      },
+    };
   });
 
-  getExtraKnobs(data);
+  return argTypes;
+};
+
+const applySpecs = (data, args) => {
```

Comment:
With apologies for the change of mind, but this function is `prepareData()` rather than applying specs. I didn't pay attention in an earlier comment what's happening in the function.